### PR TITLE
Update gnostic dependency with gnostic-models

### PR DESCRIPTION
**Pull request by liangyuanpeng**
Related:
- https://github.com/kubernetes/kubernetes/issues/118340

This PR changed:

- update go version to 1.20
- bump `k8s.io` packages to v0.28.4
- bump `sigs.k8s.io/kustomize` to v5.2.1
- bump `sigs.k8s.io/controller-runtime` to v0.16.3
- bump `sigs.k8s.io/yaml` to v1.4.0
- migrate from `google/gnostic` to `google/gnostic-models`

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,27 +1,27 @@
...
-	k8s.io/apimachinery v0.26.0
-	k8s.io/cli-runtime v0.26.0
-	k8s.io/client-go v0.26.0
-	k8s.io/component-base v0.26.0
-	k8s.io/klog/v2 v2.80.1
-	k8s.io/kubectl v0.26.0
-	k8s.io/utils v0.0.0-20230115233650-391b47cb4029
-	sigs.k8s.io/controller-runtime v0.14.1
-	sigs.k8s.io/kustomize/kyaml v0.13.9
+	k8s.io/api v0.28.0-alpha.4
```
Comment (liangyuanpeng) [inline go.mod:14]: Now, i just use  v0.28.0-alpha.4 to fix it, and i think we need to waiting for a stable version.

**Review by liangyuanpeng — COMMENTED** [review]
Some api have changed, need update.

**Comment by liangyuanpeng** [comment]
cc @seans3 @Liujingfang1 @mortent @karlkfi

**Comment by stefanprodan** [comment]
@liangyuanpeng could you please revive this PR. We really need this merged to unblock the Flux upgrade to Kubernetes 1.28. Thank you!

**Comment by ash2k** [comment]
/lgtm

@seans3 PTAL.

**Review by seans3 — COMMENTED** [review]
/lgtm

I believe you can even update to the latest `v0.28.4` (instead of `v0.28.0`) if you prefer.

@karlkfi Is this update OK with you?

**Comment by liangyuanpeng** [comment]
@ash2k @seans3   Update to v0.28.4 and not v0.28.0, PTAL,Thanks!

**Review by stefanprodan — COMMENTED** [review]
To match Kubernetes 1.28 these could also be bumped:

```
	sigs.k8s.io/controller-runtime v0.16.3
	sigs.k8s.io/kustomize/kyaml v0.15.0
	sigs.k8s.io/yaml v1.4.0
```

For reference https://github.com/fluxcd/cli-utils/pull/1/files

**Comment by liangyuanpeng** [comment]
@stefanprodan Thanks for remind.
You are the end user and have already applied this change to fluxcd, so I followed your advice and updated. Since the CI is green without this change, it would be great if you could provide more information to increase the reviewer's confidence.

Sorry about duplicate lgtm.

File: go.mod [hunk 2]
```diff
@@ -73,28 +73,30 @@ require (
...
 	google.golang.org/appengine v1.6.7 // indirect
-	google.golang.org/protobuf v1.30.0 // indirect
+	google.golang.org/protobuf v1.31.0 // indirect
 	gopkg.in/inf.v0 v0.9.1 // indirect
 	gopkg.in/yaml.v2 v2.4.0 // indirect
-	k8s.io/kube-openapi v0.0.0-20230501164219-8b0f38b5fd1f // indirect
+	k8s.io/kube-openapi v0.0.0-20230717233707-2695361300d9 // indirect
 	sigs.k8s.io/json v0.0.0-20221116044647-bc3834ca7abd // indirect
-	sigs.k8s.io/kustomize/api v0.13.2 // indirect
+	sigs.k8s.io/kustomize/api v0.13.5-0.20230601165947-6ce0bf390ce3 // indirect
```
Comment (stefanprodan) [inline go.mod:100]: Please bump this to `sigs.k8s.io/kustomize/api v0.15.0` to match `sigs.k8s.io/kustomize/kyaml v0.15.0`.

**Review by stefanprodan — APPROVED** [review]
LGTM

Thanks @liangyuanpeng

**Comment by ash2k** [comment]
@seans3 @karlkfi I think this is ready for merging, PTAL.

**Comment by liangyuanpeng** [comment]
Thanks @mortent  for approve and thanks @ash2k  help to push it.

**Pull request merged**